### ISSUE TITLE
[iOS] Extra silent accessibility focus before first ListView item

### ISSUE THREAD

**Issue by luomo-pro** [flutter/flutter#164483]
### Steps to reproduce

In Flutter master version 3.30.0-1.0.pre.428, I've identified a regression in iOS accessibility behavior. When using a ListView, there is now an extra, silent accessibility focus that appears before the first item in the list. This empty focus point doesn't announce anything when navigated to with VoiceOver, creating a confusing experience for blind users.
This issue was not present in versions from approximately 6 months ago. It appears to be a regression introduced in recent builds.

## Steps to Reproduce
1. Run the minimal example code below on an iOS device
2. Enable VoiceOver on the device
3. Navigate to the list
4. Observe that when swiping right to move through list items, there's a silent focus before the first item

### Expected results

VoiceOver should focus directly on the first item in the ListView when navigating to it.

### Actual results

When using VoiceOver, an empty, silent accessibility focus appears before the first ListView item. This focus point doesn't announce anything, and users must swipe again to reach the actual first item.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'VoiceOver Issue Demo',
      home: const AccessibilityIssueDemo(),
    );
  }
}

class AccessibilityIssueDemo extends StatelessWidget {
  const AccessibilityIssueDemo({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('VoiceOver Issue Demo'),
      ),
      body: ListView.builder(
        itemCount: 20,
        itemBuilder: (context, index) {
          return ListTile(
            title: Text('Item ${index + 1}'),
          );
        },
      ),
    );
  }
}
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel master, 3.30.0-1.0.pre.428, on macOS 14.4.1 23E224
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @luomo-pro 
I tried on latest master and stable versions running on iOS 15 device but was unable to replicate the reported behavior. The focus moves to item 1 from the app title properly. If you could provide us a short video will be helpful further along with device on which you are seeing this behavior.

**Comment by darshankawar** [comment]
Thanks for the video. I was able to replicate this on latest master and stable versions.

```
stable : 3.29.0
master : 3.30.0-1.0.pre.467
```

**Comment by zywek123** [comment]
I've mentioned it here: #104742

**Comment by chunhtai** [comment]
Scroll bar should not be exposed in semantics tree. I think this is the UIScrollView that we create to interact with iOS that received the focus.

**Comment by chunhtai** [comment]
while this is a regression, I don't think this is a very serious issue since it is not announcing incorrect information but only introduce an extra swipe.

**Comment by zywek123** [comment]
@chunhtai Why it should not be exposed in semantics tree? if It would work as a slider, Blind people could scroll the scroll views easier than using e.g. three fingers.

**Comment by luomo-pro** [comment]
1. If the scrollbar could properly support accessibility, then it could be exposed, but currently it doesn't support it at all.
   Even if exposed, I think we should provide a property to control whether it's exposed or not.
   Because some lists might only have a few items and don't need a scrollbar at all.

2. Besides ListView, I've found that TabBarView also has this focus issue.
   I feel that all scrollable views have this additional focus point.
   From the perspective of the simplest solution, removing it from the semantics tree is still the best approach.

3. In terms of severity of accessibility issues, this indeed isn't that serious.
   However, for blind users, it reduces browsing efficiency and feels awkward.
   It's like sighted users seeing an extra button in the interface that doesn't do anything - it's uncomfortable.
   It also causes confusion for blind users - they'll question what the purpose of this focus point is.

**Comment by cuberob** [comment]
Also seeing this issue unfortunately. Hope this can be picked up soon, as it's quite confusing for users when they perform a scroll event that results in zero feedback.

Also the timing is magnificent, we are just about to do an accessibility audit of our app, haha, Murphy's law! Time for a temporary flutter downgrade I guess...!

**Comment by chunhtai** [comment]
> Why it should not be exposed in semantics tree? if It would work as a slider, Blind people could scroll the scroll views easier than using e.g. three fingers.

interactable scrollbar is mostly a desktop and web things, and they are not interactable by screen reader in native app. I think we should follow the same behavior.

User can three fingers scroll Flutter app already, it is a separate feature from the scrollbar.

**Comment by zywek123** [comment]
@chunhtai If you have an iPhone or some Apple devices, open any application, and at the right side of the screen VoiceOver can see Scroll bar, Set the focus there and swipe up or down, You will see the screen scrolls to the desired position. You don't have to use three fingers to scroll.

**Comment by luomo-pro** [comment]
@loic-sharma
I've upgraded to the latest master version (3.31.0-1.0.pre.284), and the issue still exists.

**Comment by chunhtai** [comment]
@zywek123 
I think we are off topic since this issue is about extra swipe to focus list item. Can you create a new issue for scrollbar semantics so that we can continue with the discussion

**Comment by ddfreiling** [comment]
Any ETA on getting this fixed? Bit of a show-stopper for updating a screen-reader focused app of ours.

**Comment by zywek123** [comment]
@ddfreiling probably not in next release, i've tested it yesterday on latest master and the bug still occurs.

**Comment by Tryneeth** [comment]
Any news about this?

**Comment by AnnaPS** [comment]
I'm also facing this problem, any news on this?

**Comment by mahmoudelbanna** [comment]
I'm also encountering this issue. It seems to be the same root cause as [#99011](https://github.com/flutter/flutter/issues/99011) — where a FlutterSemanticsScrollView node is automatically added by the iOS platform embedding, leading to an extra accessibility focus before the first visible item in a scrollable.

Is there any update on this? It appears to be a long-standing issue and affects the accessibility experience significantly.

Tested on:

- iOS 18.5

- Flutter version 3.32.4

**Comment by kykbmvkgj8686755-gif** [comment]
I also encountered this issue. May I ask if there is a solution now?

**Comment by BLKKKBVSIK** [comment]
This is a long persisting issue that today is making a lot of damage for accessibility audit (WCAG / RAAM at least).
Our accessibility score declined because of this issue, and unfortunately we still have no good solution for it.

**Comment by chunhtai** [comment]
Some insight on this issue. This extra swipe is here mainly due to a workaround that iOS does not expose API to annotate a UIAccessibility to be a scroll view.

We use a real UIScrollView to interact with iOS, this UIScrollView however has to be focusable otherwise other assistive technologies like switch control or voicecontrol can't interact with it. You will see the same behavior if you are building a UIScrollView in a native iOS app.

I am not sure whether things have changed since I introduced this, but without this other assistive technologies may break.

**Comment by chunhtai** [comment]
If we can' figure this out, we can instead use UITableView to interact with iOS, the problem with that approach is that it only works on vertical scrollable and the class is also harder to maintain.

**Comment by zywek123** [comment]
@chunhtai It will be enough to hide this Semantic node as it was before flutter 3.29 as a temporary solution.

**Comment by chunhtai** [comment]
hmmm I thought it is auto hidden already https://github.com/flutter/flutter/blob/560a50038074d6a93eb2844990e7b8f41b1aedcd/engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm#L42

Something else may be going on

**Comment by zywek123** [comment]
@chunhtai Should it be hidden in 3.35.4? If not, I will try to check it in the morning.

**Comment by vintage** [comment]
@zywek123 Any luck? The last known version of Flutter where it worked correctly for me was `3.27.4`. Unfortunatelly I couldn't hold upgrading the framework any longer in my projects, so at the moment visually impaired users have trouble navigating smoothly through the app.

**Comment by zywek123** [comment]
@vintage I have no idea, stay on 3.27 or update. We updated and well, blind people have to deal with this situation. You can't work on outdated Flutter forever.
@chunhtai [✓] Flutter (Channel main, 3.37.0-1.0.pre-388, on macOS 15.5 24F74 darwin-arm64, The scroll bar is still visible to VO, not only before the first ListView element, but also before the first and between GridView elements if they are not tightly adjacent to each other.

**Comment by jimgerth** [comment]
I can reproduce this in Flutter `3.41.2` on an actual iOS device.
